Source/Scene/Multiple3DTileContent.js

```diff
@@ -0,0 +1,540 @@
+//import defaultValue from "../Core/defaultValue.js";
+import defined from "../Core/defined.js";
+import destroyObject from "../Core/destroyObject.js";
+import Request from "../Core/Request.js";
+import RequestScheduler from "../Core/RequestScheduler.js";
+import RequestState from "../Core/RequestState.js";
+import RequestType from "../Core/RequestType.js";
+import RuntimeError from "../Core/RuntimeError.js";
+import when from "../ThirdParty/when.js";
+import Cesium3DTileContentType from "./Cesium3DTileContentType.js";
+import Cesium3DTileContentFactory from "./Cesium3DTileContentFactory.js";
+import preprocess3DTileContent from "./preprocess3DTileContent.js";
+
+/**
+ * A collection of contents for tiles that use the <code>3DTILES_multiple_contents</code> extension.
+ * <p>
+ * Implements the {@link Cesium3DTileContent} interface.
+ * </p>
+ *
+ * @see {@link https://github.com/CesiumGS/3d-tiles/tree/3d-tiles-next/extensions/3DTILES_multiple_contents/0.0.0|3DTILES_multiple_contents extension}
+ *
+ * @alias Multiple3DTileContent
+ * @constructor
+ *
+ * @param {Cesium3DTileset} tileset The tileset this content belongs to
+ * @param {Cesium3DTile} tile The content this content belongs to
+ * @param {Resource} tilesetResource The resource that points to the tileset. This will be used to derive each inner content's resource.
+ * @param {Object} extensionJson The <code>3DTILES_multiple_contents</code> extension JSON
+ *
+ * @private
+ */
+export default function Multiple3DTileContent(
+  tileset,
+  tile,
+  tilesetResource,
+  extensionJson
+) {
+  this._tileset = tileset;
+  this._tile = tile;
+  this._tilesetResource = tilesetResource;
+  this._contents = [];
+
+  var contentHeaders = extensionJson.content;
+  this._innerContentHeaders = contentHeaders;
+  this._requestsInFlight = 0;
+
+  /**
+   * This is set to true if one of the inner contents are canceled.
+   * @type {Boolean}
+   * @private
+   */
+  this.canceled = false;
```

Comment:
I might be missing it, but is this ever set to `true`?

Reply:
:open_mouth: you're right, I'm missing a line. Though now one of my tests is hanging... I'll have to do some investigation.

Reply:
Ah I see, it's a race condition, multiple tiles are canceled, but the tile is only reset once. I think I can fix this with a second variable to keep track of how many times the content request has been attempted. I think we mentioned this before but I wasn't sure that I needed it until now.

Reply:
okay there's more going on here than I thought... basically I'm not refreshing the request objects, so once canceled they get stuck there. I need to think through this a bit, in tandem with https://github.com/CesiumGS/cesium/pull/9407#discussion_r592505966